### DISCUSSION TITLE
Examples of good reference docs (from anywhere)?

### DISCUSSION THREAD

**Discussion by EricZinda** [delph-in/docs#105] [Team Posts]
(I don't know if this works, but) @delph-in/erdw , this thread is for collecting good examples of reference documentation, even if it is just one part of something.  Please post and describe what we might want to copy/emulate!

**Comment by lizcconrad** [comment]
This is a little more general than an example, but this is the "documentation paradigm" I mentioned in one of the discussions: https://documentation.divio.com/. It seems very effective and the central point of it is that you must consider the use case/audience you are writing for every time you write a piece of documentation, which I think this group will need to keep in mind since we discussed how certain pages are going to be catered to different audiences.

**Comment by EricZinda** [comment]
I love that taxonomy! I'm going to start adopting in our discussions to help clarify the focus of the different things we write.

**Comment by arademaker** [comment]
@lizcconrad, excellent link, and material...

**Comment by EricZinda** [comment]
I worked on the .Net Framework so I thought I'd see if they have anything we can copy.

[The top level overview page](https://docs.microsoft.com/en-us/dotnet/) clearly directs people towards the different categories: Tutorials (Create your application), HowTos (Develop .NET apps), Explanation (.NET concepts), Reference (API and language reference).

The docs seem to broken into main "sites" that have independent table of contents trees on the left side:
- [Conceptual and HowTo](https://docs.microsoft.com/en-us/dotnet/fundamentals/)
- [API Reference](https://docs.microsoft.com/en-us/dotnet/api/?view=net-6.0)
And lots of other smaller ones like: [Azure for mobile developers](https://docs.microsoft.com/en-us/azure/developer/mobile-apps/), [Game Development for visual studio](https://visualstudio.microsoft.com/vs/features/game-development/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link), etc.

## Explanation Docs
Explanation docs are not versioned, which makes sense since they are conceptual and the high level concepts don't change all that much.  I assume they have notes in them about version differences if there were any.

If you look at a conceptual topic like [Assemblies in .Net](https://docs.microsoft.com/en-us/dotnet/standard/assembly/) it has some Explanation and some simple Tutorials in in.

## Reference Docs
Reference docs are versioned.  The entire TOC on the left refers to a particular version and you have to choose which you look at.

They have a [good writeup on the Reference docs experience](https://docs.microsoft.com/en-us/teamblog/announcing-unified-dotnet-experience-on-docs) that talks about their "Reference Docs browser" works, how they do versioning, searching, etc.

If you look at a particular [reference topic like `system.string`](https://docs.microsoft.com/en-us/dotnet/api/system.string?view=netframework-4.8), you get a *ton* of information. I'd expect this for one of the most used (and most complicated since it has to do with language) types in the system.  At the top is the basic reference stuff, and then farther down are lots of examples and how tos, and some conceptual stuff.  You can see the version you are looking at on the left and change it.

## Search
Doing a search using the text box on the upper right corner searches across *everything* as far as I can tell.  If you do a search for "system.string" (which would be like searching for something like `_the_q`, I guess), you get hits across all types of documentation that you can filter on the left:

![image](https://user-images.githubusercontent.com/2416921/186480317-699cc4f0-2f7e-4caa-87cc-8e8e85093e5a.png)

As far as I can tell, it only searches against the *latest* version of the reference docs though.  If you click a `system.string` result, you'll see the latest docs for it. You'll have to pick the dropdown version of another version when you get there if you want something different.

## Take-aways
I like using the basic model of:
- Conceptual docs are not versioned (but might have version specific notes in them)
- Reference docs are versioned and have a way to switch between versions
- Conceptual docs and Reference docs have different TOC trees for navigation (i.e. feel like different subsites)
- Search searches all conceptual docs and the current version of reference docs, with ways to filter the results to either if you want
